pennylane/math/single_dispatch.py

```diff
@@ -122,14 +123,27 @@ def _permutation_parity(perm):
                 cycle_length += 1
 
             if cycle_length:
-
                 parity *= (-1) ** (cycle_length - 1)
     return parity
 
 
 ar.register_function("scipy", "linalg.det", _det_sparse)
 ar.register_function("scipy", "linalg.eigs", sp.sparse.linalg.eigs)
+# ar.register_function("scipy", "linalg.eigh", sp.sparse.linalg.eigh)

```

Comment:
```suggestion
```
No method `eigh` in `scipy.sparse.eigh`

Reply:
Isn't https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.linalg.eigsh.html the sparse equivalent? Can we bind to this?

Reply:
@mlxd `np.linalg.eigh` will precisely calculate all the eigs of the given square matrix. However, `scipy.sparse.linalg.eigsh` can only take limited number of suspaces; they denoted that "k must be smaller than N". This is also a quite confusing fact to me and I would like to know why the package was implemented in such a way...

Reply:
This is how the algorithm works at the FORTRAN layer (the Python layer binds to https://github.com/opencollab/arpack-ng). It relies on the [Lanzcos alg](https://en.wikipedia.org/wiki/Lanczos_algorithm) under the hood, which nominally gives you a set of "useful" eigenvals/vecs, rather than all. They accept an argument to choose looking for the largest, smallest, etc using the `which` argument. It's fine to expect this is how it works in practice, though if you want to use the values, controlling what is returned may be worth modifying the function signature to accommodate the control offered at the scipy layer.

Reply:
Backlog: sc-85634